objects/src/accounts/builder/mod.rs

```diff
@@ -167,63 +158,52 @@ impl AccountBuilder {
     pub fn build(self) -> Result<(Account, Word), AccountError> {
         let (init_seed, vault, code, storage) = self.build_inner()?;
 
+        #[cfg(any(feature = "testing", test))]
+        if !vault.is_empty() {
+            return Err(AccountError::BuildError(
+                "account asset vault must be empty on new accounts".into(),
+                None,
+            ));
+        }
+
         let (account_id, seed) =
             self.grind_account_id(init_seed, code.commitment(), storage.commitment())?;
 
         debug_assert_eq!(account_id.account_type(), self.account_type);
         debug_assert_eq!(account_id.storage_mode(), self.storage_mode);
 
-        let account = Account::from_parts(account_id, vault, storage, code, self.nonce);
+        let account = Account::from_parts(account_id, vault, storage, code, Felt::ZERO);
 
         Ok((account, seed))
     }
 }
 
 #[cfg(any(feature = "testing", test))]
 impl AccountBuilder {
-    /// Sets the nonce of the account. This method is optional.
-    ///
-    /// If unset, the nonce will default to [`ZERO`].
-    pub fn nonce(mut self, nonce: Felt) -> Self {
-        self.nonce = nonce;
-        self
-    }
-
     /// Adds all the assets to the account's [`AssetVault`]. This method is optional.
     ///
-    /// Must only be called when nonce is non-[`ZERO`] since new accounts must have an empty vault.
+    /// Must only be called when nonce is non-[`Felt::ZERO`] since new accounts must have an empty
+    /// vault.
     pub fn with_assets<I: IntoIterator<Item = Asset>>(mut self, assets: I) -> Self {
```

Comment:
Is this part of the comment still relevant? (seems like we are always setting nonce to `ONE` for existing accounts).

Reply:
Rephrased it. It's still possible to use the method with `build` so some comment seems necessary.